specs/multiple_node_ids_in_jobs.adoc

```diff
@@ -0,0 +1,119 @@
+= Pass multiple node-ids in a job submitted to central store job queue
+
+Currently a job can be targeted to a specific node by explicitly setting the
+attribute `node_id` in job structure. The matching the node-id picks up and
+processes the job from central store job queue.
+
+We should be able to submit job targeted to a set of nodes rather submitting to
+a specific node. This feature would implicitly enable targeting of a job to
+specific node anyway.
+
+== Problem description
+
+The job targeted to a single node restricts user and also it becomes a single
+point of failure. Additionally in case of `ceph` certain jobs can be and should
+be picked of any of the mon nodes in the cluster and we should not be
+restricting to a specific node.
+
+User should be able to pass set of node-ids in job and any one of the storage
+nodes underlying should pick and process this job.
+
+== Use Cases
+
+* Job execution by storage nodes from central store job queue
+
+== Proposed change
+
+* Enhance the job execution modules in Tendrl to looks for the job-ids list from
+the fetched job out of central store job queue, and process only if current
+node's id matches one in the list
+
```

Comment:
But how can this handle the case of "any one of the" as briefed in the "Problem Description" section in "User should be able to pass set of node-ids in job and any one of the storage nodes underlying should pick and process this job."

I feel the case of "any one of the" should be made by the application loading the job and the case of execute by one/or set of nodes will be handled as you described in "Proposed Change". Is my understanding correct?